### DISCUSSION TITLE
Is there any way to postion the blocks ?

### DISCUSSION THREAD

**Discussion by stan900814** [codex-team/editor.js#1890] [Q&A]
I'm  foreach the data  form the save( ).and create a list depend on this data.text . now i want to add a click event on this list's item,
it can link to the editor's blocks and focus the block of  i clicked . 
is there any way to get it ?

**Comment by neSpecc** [comment]
And what's the problem?

**Comment by sos-productions** [comment]
What I did  (assuming the editor id is 'editorjs')  to postion the the block pos indexed n=pos-1 as the first block is 0:

//we get the redactor's dom holding the blocks more universally than document.getElementsByClassName("codex-editor__redactor");)  
const edom=window.document.getElementById('editorjs');
let blocks=edom.firstChild.firstChild.children; 

//Hook as editor.caret.setToBlock('start', index);editor.caret.focus() does not work
blocks[n].scrollIntoView(true);  
 blocks[n].focus();

However I am stuck too to add a mouseover handler, there is already a click event registered in ui.ts so I guess it is not possible to add one from outside.

   var redactor=document.getElementsByClassName("codex-editor__redactor");
//the next lines produces Unhandled promise rejection TypeError: "t.addEventListener is not a function" wonder why
          editor.listeners.on.on(redactor,"mouseover",function(e){ 
                    console.log(this);
           },false);

**Comment by sos-productions** [comment]
Update that works to retrieve the index of the block clicked..I did not know the difference between HTMLCollection live nodes and static NodeList.. This fix the error previously works for mouseover also.

  function getRedactor() {
                //console.log(document.getElementsByClassName("codex-editor__redactor")); //Live, HTMLCollecton will not work
                return document.querySelectorAll('.codex-editor__redactor')[0]; //NodeList that contains static NodeList usable.
           }
           
           function getBlocksNodeList() {
                var redactor=getRedactor();
                return redactor.childNodes; //NodeList <div class="ce-block" or class="ce-block focused" 
           }
           
           var blocks = getBlocksNodeList();
           
            for(var i = 0, len = blocks.length ; i < len ; i++) {
                blocks[i].addEventListener('click', function(e){
                        var n=Array.prototype.indexOf.call(blocks, this); 
                        //Do your stuff!
                        console.log(n);
                })
            }
